### PR TITLE
Bugfix/2 comments from david

### LINKED ISSUE #2: comments from david
**Should Fix**

- [x]  Within the TimeplusSinkTask various OkHttp responses opened and not closed client.newCall(request).execute(); alongside some potential Null Pointers for response.body().string()
- [ ] Would recommend looking into https://docs.confluent.io/platform/current/connect/security.html#externalize-secrets to externalise secrets outside of the connect configuration and avoid passing raw API strings around in payloads.
https://rmoff.net/2019/05/24/putting-kafka-connect-passwords-in-a-separate-file-/-externalising-secrets/ 
- [x] Error handling rather than use of console logs (and ideally use a logging framework) so the user can get error feedback and potential use of DLQ patterns to allow for replying messages https://www.confluent.io/en-gb/blog/kafka-connect-deep-dive-error-handling-dead-letter-queues/
- [x] Back off strategies (something like [Resilience4j](https://resilience4j.readme.io/docs) will slot into OkHttp and will prevent downstream issues if the HTTP server cannot keep up
- [ ] https://docs.confluent.io/platform/current/connect/references/allconfigs.html#cp-license-configuration
- [x] Unit Tests for the code so far; 
- [x] ideally Integration tests  ([testcontainers](https://www.testcontainers.org/modules/kafka/) is a good resource to bring up a sample kafka cluster)
- [x] Technical support details added to the manifest file and the readme; so anyone can know where to get support if there is an issue in the future.

**Could Fix**

- [ ] Generally we find customers like to have more supported data types than String; and use JSON or AVRO types) https://docs.confluent.io/platform/current/streams/developer-guide/datatypes.html#kstreams-data-types-and-serialization
Supporting the schema registry https://docs.confluent.io/platform/current/schema-registry/index.html

- [ ] Setting error tolerance for failures https://www.confluent.io/en-gb/blog/kafka-connect-deep-dive-error-handling-dead-letter-queues/#:~:text=tolerance%20%3D%20all%20will%20enable%20Kafka,message%20failures%20that%20do%20occur.

**Recommendations for Fixes**

- [ ] Use of validator and recommenders will avoid potential issues with poor configuration settings (such as raw or json) https://www.confluent.io/en-gb/blog/write-a-kafka-connect-connector-with-configuration-handling/

### PR DESCRIPTION
**Pull request by gangtao**
- add response.close()
- add retry for server side error to support flow control
- add logging
- add ut
- add it

### CODE REVIEW

File: src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java [hunk 1]
```diff
@@ -132,7 +160,7 @@ private String getCreateJSONPayload(String stream, String event) {
         return payload.toString();
     }
 
-    private void createStream(String event) {
+    void createStream(String event) {
```
Comment (ye11ow) [inline src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java:157]: is it common practise for a kafka connector to create resource?

File: src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java [hunk 2]
```diff
@@ -90,21 +98,41 @@ public void put(Collection<SinkRecord> records) {
...
 
+        client.interceptors().add(new Interceptor() {
+            @Override
+            public Response intercept(Chain chain) throws IOException {
+                Request request = chain.request();
+
+                // try the request
+                Response response = chain.proceed(request);
+                int tryCount = 0;
+                while ((response.code() == 429 || response.code() >= 500) && tryCount < MAX_RETRY_COUNT) {
```
Comment (ye11ow) [inline src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java]: Shall we just use an existing java lib for retry? e.g. https://failsafe.dev/

File: src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java (hunk 1 above)
Comment (gangtao) [reply src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java:157]: not all sink need resource creation, it really depends, in our case, it is optional when you want to try it with minial manual work

File: src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java (hunk 2 above)
Comment (gangtao) [reply src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java]: we can use, but I not sure which lib is better to use, any recommendation?

Comment (gangtao) [reply src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java]: using failsafe now

**Review by jovezhong — APPROVED** [review]

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**